python/ray/dashboard/modules/serve/serve_head.py

```diff
@@ -165,6 +165,46 @@ async def put_all_applications(self, req: Request) -> Response:
         else:
             return Response()
 
+    @routes.get(
+        "/api/serve/applications/{application_name}/deployments/{deployment_name}/scale"
+    )
+    @dashboard_optional_utils.init_ray_and_catch_exceptions()
+    @validate_endpoint()
+    async def get_deployment_scale(self, req: Request) -> Response:
+        application_name = req.match_info["application_name"]
+        deployment_name = req.match_info["deployment_name"]
+
+        controller = await self.get_serve_controller()
+        if controller is None:
+            return Response(
+                status=404,
+                text="No Serve instance running on this Ray cluster.",
+            )
+
+        try:
+            num_replicas = await controller.get_deployment_num_replicas.remote(
+                deployment_name, application_name
+            )
```

Comment:
can we use `controller.get_deployment_details` instead of creating a new function ?